### PR TITLE
Restructured test utility functions

### PR DESCRIPTION
**Pull request by CGodiksen**
This PR restructures the utility functions used during testing so they are all available in the same place within the `modelardb_common` crate. Previously we had utility functions in `modelardb_common`, `modelardb_common_test`, and `common_test.rs`. `modelardb_common_test` and `common_test.rs` have now been removed and the functionality in them has either been moved to `modelardb_common` or replaced by simple test utility functions in the relevant files directly.

In order to make it easier to set up a `Context` during testing, a `try_new()` method has also been added to `Context`.

### CODE REVIEW

File: crates/modelardb_server/src/optimizer/model_simple_aggregates.rs [hunk 1]
```diff
@@ -1065,6 +1068,78 @@ mod tests {
             vec![TypeId::of::<ParquetExec>(), TypeId::of::<ParquetExec>()],
         ];
 
-        common_test::assert_eq_physical_plan_expected(physical_plan, &expected_plan);
+        assert_eq_physical_plan_expected(physical_plan, &expected_plan);
+    }
+
+    /// Parse, plan, and optimize the `query` for execution on data in `path`.
+    pub async fn query_optimized_physical_query_plan(
```
Comment (skejserjensen) [inline crates/modelardb_server/src/optimizer/model_simple_aggregates.rs:1075]: Unclear why `query_optimized_physical_query_plan()` is still public after being moved to `model_simple_aggregates.rs`, maybe either make it private or add a comment or doc comment.

**Review by skejserjensen — CHANGES_REQUESTED** [review]

File: crates/modelardb_common/Cargo.toml [hunk 2]
```diff
@@ -29,6 +29,7 @@ sqlparser.workspace = true
 sqlx = { workspace = true, features = ["runtime-tokio-rustls"] }
 tonic.workspace = true
 uuid.workspace = true
+rand.workspace = true
```
[the remark this answers is not in the document]
Comment (CGodiksen) [reply crates/modelardb_common/Cargo.toml:32]: Fixed.

File: crates/modelardb_server/src/context.rs [hunk 3]
```diff
@@ -51,6 +57,82 @@ pub struct Context {
...
 impl Context {
+    /// Create the components needed in the [`Context`] and use them to create the [`Context`]. If
+    /// a metadata manager or storage engine could not be created, [`ModelarDbError`] is returned.
+    pub async fn try_new(
+        runtime: Arc<Runtime>,
+        data_folders: &DataFolders,
+        cluster_mode: ClusterMode,
+        server_mode: ServerMode,
+    ) -> Result<Self, ModelarDbError> {
+        // Create the components for the context.
```
[the remark this answers is not in the document]
Comment (CGodiksen) [reply crates/modelardb_server/src/context.rs:68]: Fixed.

File: crates/modelardb_server/src/optimizer/model_simple_aggregates.rs (hunk 1 above)
Comment (CGodiksen) [reply crates/modelardb_server/src/optimizer/model_simple_aggregates.rs:1075]: Good catch, it has been made private.

File: crates/modelardb_server/src/optimizer/model_simple_aggregates.rs [hunk 4]
```diff
@@ -1065,6 +1068,78 @@ mod tests {
...
+            .await
+            .unwrap()
+            .create_physical_plan()
+            .await
+            .unwrap()
+    }
+
+    /// Assert that `physical_plan` and `expected_plan` contain the same operators. `expected_plan`
+    /// only contains the type ids so the tests do not have to construct the actual operators.
+    pub fn assert_eq_physical_plan_expected(
```
[the remark this answers is not in the document]
Comment (CGodiksen) [reply crates/modelardb_server/src/optimizer/model_simple_aggregates.rs:1123]: Good catch, it has been made private.

File: crates/modelardb_server/src/context.rs [hunk 5]
```diff
@@ -51,6 +57,82 @@ pub struct Context {
...
+        let metadata_manager = Arc::new(
+            MetadataManager::try_new(&data_folders.local_data_folder)
+                .await
+                .map_err(|error| {
+                    ModelarDbError::ConfigurationError(format!(
+                        "Unable to create a MetadataManager: {error}"
```
[the remark this answers is not in the document]
Comment (CGodiksen) [reply crates/modelardb_server/src/context.rs:74]: I also do not have any strong opinions but I think i lean towards just using ": {error}". I have created a low priority [work item](https://dev.azure.com/ModelarData/ModelarDB%20Implementation/_workitems/edit/794) to handle this.

**Review by skejserjensen — APPROVED** [review]

**Review by chrthomsen — APPROVED** [review]

**Pull request merged**